lib/OpenQA/Shared/Controller/OpenAPI.pm

```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Shared::Controller::OpenAPI;
+use Mojo::Base 'Mojolicious::Controller', -signatures;
+use Mojo::JSON qw(encode_json decode_json);
+
+sub catchall ($c) {
+    my $result = $c->validate_request;
+    unless ($result->valid) {
+        return $c->render(status => 400, json => {result => $result});
+    }
+    my $VALIDATE = $c->stash('openapi');
+    my $path_captures = $VALIDATE->{path_captures};
+    $c->stash($_, $path_captures->{$_}) for keys %$path_captures;
+    my $op = $VALIDATE->{operation_id};
+    if ($op =~ s/^operation_(jobs|groups)_//) {
+        return $c->$op();
+    }
```

Comment:
I'm wondering whether Mojolicious provides some kind of hook so we can still use its logic to call the right function instead of having to reimplement it. Then we could probably also avoid changes to the route registrations.

Reply:
The point is that the openAPI document provides the logic for the right function. We don't need our route registrations anymore, as it would be duplicated logic.